editor/src/components/canvas/commands/wrap-in-conditional-command.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  emptyComments,
+  jsExpressionValue,
+  jsxConditionalExpression,
+  JSXElementChild,
+  jsxFragment,
+} from '../../../core/shared/element-template'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import {
+  EditorState,
+  EditorStatePatch,
+  forUnderlyingTargetFromEditorState,
+  insertElementAtPath,
+  removeElementAtPath,
+} from '../../editor/store/editor-state'
+import { BaseCommand, CommandFunction, getPatchForComponentChange, WhenToRun } from './commands'
+import * as EP from '../../../core/shared/element-path'
+import { getUtopiaJSXComponentsFromSuccess } from '../../../core/model/project-file-utils'
+import { InsertionSubjectWrapper } from '../../editor/editor-modes'
+import { assertNever } from '../../../core/shared/utils'
+
+type ContainerToWrapIn = InsertionSubjectWrapper
+
+export interface WrapInContainerCommand extends BaseCommand {
+  type: 'WRAP_IN_CONDITIONAL'
+  whenToRun: WhenToRun
+  target: ElementPath
+  wrapperUID: string
+  wrapper: ContainerToWrapIn
+}
+
+export function wrapInContainerCommand(
+  whenToRun: WhenToRun,
+  target: ElementPath,
+  wrapperUID: string,
+  wrapper: ContainerToWrapIn,
+): WrapInContainerCommand {
+  return {
+    type: 'WRAP_IN_CONDITIONAL',
+    whenToRun: whenToRun,
+    target: target,
+    wrapperUID: wrapperUID,
+    wrapper: wrapper,
+  }
+}
+
+export const runWrapInConditionalCommand: CommandFunction<WrapInContainerCommand> = (
+  editor: EditorState,
+  command: WrapInContainerCommand,
+) => {
+  let editorStatePatches: Array<EditorStatePatch> = []
+
+  forUnderlyingTargetFromEditorState(
+    command.target,
+    editor,
+    (success, elementToWrap, _underlyingTarget, underlyingFilePath) => {
+      const components = getUtopiaJSXComponentsFromSuccess(success)
+      const withElementRemoved = removeElementAtPath(command.target, components)
+
+      const wrapper = getInsertionSubjectWrapper(command.wrapper, command.wrapperUID, elementToWrap)
+
+      // Insert the conditional at the initial index
+      const targetParent = EP.parentPath(command.target)
+      const insertionResult = insertElementAtPath(
+        editor.projectContents,
+        underlyingFilePath,
+        targetParent,
+        wrapper,
+        withElementRemoved,
+        null, // FIXME Find the index position of the original element
+      )
+
+      editorStatePatches.push(
+        getPatchForComponentChange(
+          success.topLevelElements,
+          insertionResult.components,
+          success.imports,
+          underlyingFilePath,
+        ),
+      )
+
+      const wrapperPath = EP.appendToPath(targetParent, wrapper.uid)
+
+      editorStatePatches.push({
+        selectedViews: {
+          $set: [wrapperPath],
+        },
+      })
+    },
+  )
+
+  return {
+    editorStatePatches: editorStatePatches,
+    commandDescription: `Wrapped Element ${EP.toUid(command.target)} in a conditional`,
```

Comment:
I believe this should be:
```suggestion
    commandDescription: `Wrapped Element ${EP.toUid(command.target)} in a ${command.wrapper}`,
```

Reply:
thanks! updated in https://github.com/concrete-utopia/utopia/pull/3531/commits/7b093a70c94b5a663a264a33d734f830bc2e43c6